07_IPC/03_Message_Passing.md

```diff
@@ -33,16 +33,27 @@ As mentioned, there's some initial setup that goes into message passing: we need
 Since the struct representing the endpoint is going to be accessed by multiple processes, we'll want a lock to protect the data from race conditions. We'll use the following struct to represent our endpoint:
 
 ```c
+
+typedef struct ipc_message{
+    void* message_buffer;
+    size_t message_size;
+    uintptr_t next_message;
+} ipc_message_t;
+
+ typedef struct ipc_message_queue{
+    ipc_message_t* messages;
+} ipc_message_queue_t;
+
 struct ipc_endpoint {
     char* name;
-    void* msg_buffer;
-    size_t msg_length;
+    ipc_message_queue_t* queue;
+    uint64_t owner_pid;
```

Comment:
I'd prefer `tid_t` over `uint64_t` here, it describes intent better. `uint64_t` may not always be a convenient type depending on the architecture.